src-java/northbound-service/northbound/src/main/java/org/openkilda/northbound/converter/HaFlowMapper.java

```diff
@@ -98,4 +113,70 @@ public abstract class HaFlowMapper {
+    }
+
+    @Mapping(target = "forward", source = "forwardPath")
+    @Mapping(target = "reverse", source = "reversePath")
+    @Mapping(target = "protectedPath", expression = "java(toHaFlowProtectedPath(source.getProtectedPath(), null))")
+    @Mapping(target = "diverseGroup", expression = "java(toDiverseGroupPayload(source, null))")
+    public abstract HaFlowPath toHaFlowPath(FlowPathDto source);
+
+    @Mapping(target = "forward", source = "source.forwardPath")
+    @Mapping(target = "reverse", source = "source.reversePath")
+    @Mapping(target = "diverseGroup", expression = "java(toDiverseGroupPayload(source, diverseWithFlows))")
+    public abstract HaFlowProtectedPath toHaFlowProtectedPath(FlowProtectedPathDto source,
+                                                              List<FlowPathDto> diverseWithFlows);
+
+    @Mapping(target = "flowId", source = "source.id")
+    @Mapping(target = "reverse", source = "source.reversePath")
+    @Mapping(target = "forward", source = "source.forwardPath")
+    @Mapping(target = "protectedPath",
+            expression = "java(toHaFlowProtectedPath(source.getProtectedPath(), diverseWithFlows))")
+    @Mapping(target = "diverseGroup", expression = "java(toDiverseGroupPayload(source, diverseWithFlows))")
+    public abstract HaSubFlowPath toSubFlowPath(FlowPathDto source, List<FlowPathDto> diverseWithFlows);
+
+    protected DiverseGroupPayload toDiverseGroupPayload(FlowPathDto source, List<FlowPathDto> diverseWithFlows) {
+        OverlappingSegmentsStats segmentsStats = source.getSegmentsStats();
+        if (segmentsStats != null) {
+            return DiverseGroupPayload.builder()
+                    .overlappingSegments(segmentsStats)
+                    .otherFlows(diverseWithFlows != null
+                            ? toGroupPaths(diverseWithFlows, FlowPathDto::isPrimaryPathCorrespondStat) : null)
+                    .build();
+        }
+        return null;
+    }
+
+    protected DiverseGroupPayload toDiverseGroupPayload(FlowProtectedPathDto source,
+                                                        List<FlowPathDto> diverseWithFlows) {
+        if (source != null) {
+            OverlappingSegmentsStats segmentsStats = source.getSegmentsStats();
+            if (segmentsStats != null) {
+                return DiverseGroupPayload.builder()
+                        .overlappingSegments(segmentsStats)
+                        .otherFlows(diverseWithFlows != null
+                                ? toGroupPaths(diverseWithFlows, e -> !e.isPrimaryPathCorrespondStat()) : null)
+                        .build();
+            }
+        }
+        return null;
+    }
+
+    private List<GroupFlowPathPayload> toGroupPaths(List<FlowPathDto> paths, Predicate<FlowPathDto> predicate) {
+        return paths.stream()
+                .filter(predicate)
+                .map(pathMapper::mapGroupFlowPathPayload)
+                .collect(Collectors.toList());
+    }
```

Comment:
Could you please add unit tests for the mapper?

Reply:
Sure, will add.